modules/kubernetes-addons/prometheus/main.tf

```diff
@@ -109,7 +109,7 @@ module "irsa_amp_ingest" {
   kubernetes_namespace        = local.namespace
 
   kubernetes_service_account    = local.ingest_service_account
-  irsa_iam_policies             = [aws_iam_policy.ingest[0].arn]
+  irsa_iam_policies             = [try(aws_iam_policy.ingest[0].arn, "")]
```

Comment:
Not sure I understand this one.

Reply:
I am not sure if this is the best solution, but right now if you are trying to cleanup the complete-k8s addon for example, you'll face the failure explained in the details of the PR where `aws_iam_policy.ingest is empty tuple` , when you try to the delete the resource (that was already deleted), because it's under a condition (if mng prometheus is enabled), this was the cleanest workaround I found. Terraform is trying to use the resource because it's also being used in the [rest of the prometheus module](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/modules/kubernetes-addons/prometheus/main.tf#L12) and because it has that count condition

Reply:
it fixes an error so for now we'll go with it - we can revisit later to see if there is a better or more idiomatic way to resolve this